amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/DeterministicShuffleShardSyncLeaderDecider.java

```diff
@@ -68,18 +68,22 @@
      * @param leaderElectionThreadPool    Thread-pool to be used for leaderElection.
      * @param numPeriodicShardSyncWorkers Number of leaders that will be elected to perform periodic shard syncs.
      */
-    DeterministicShuffleShardSyncLeaderDecider(LeaseRefresher leaseRefresher, ScheduledExecutorService leaderElectionThreadPool,
-            int numPeriodicShardSyncWorkers) {
+    DeterministicShuffleShardSyncLeaderDecider(LeaseRefresher leaseRefresher,
```

Comment:
We need a new constructor that allows to take in the dedicated arg. We should not change constructors for public classes.

Reply:
Sounds good. Added a new constructor